### ISSUE TITLE
Support default values for hyperparameters

### ISSUE THREAD

**Issue by dsherry** [alteryx/evalml#353]
**Problem**
Currently, our tuners aren't set up to suggest default values as the starting point for a given pipeline's hyperparameter search. I think we should provide this functionality.

My justification is as follows. We could choose default values for component parameters through testing, or through intuition. In either case, having the parameter search begin with values we think are most generally applicable will be beneficial.

Further, some of the third-party modeling libraries we use have functionality for internally choosing intelligent defaults, either as a constant or via other internal heuristics. By providing an external value for those parameters, we may produce lower performance. We should have a mechanism for using the third-party libraries' intelligent defaults as the starting point, even if we support search on the parameter in question.

Ex: on the #247 catboost PR, we expose the learning rate as a hyperparameter for search ("eta"), but catboost has internal functionality for choosing a default learning rate.

**Proposal**
1. Let's augment the Tuner API and hyperparameter values to support default values. Perhaps we can update the datastructure for each hyperparameter, like this:
```
class MyClassifier(Estimator):
    hyperparameter_ranges = {
        "a_model_parameter": {"values": ["option_a", "option_b], "default": "option_a"}
    }
```

An alternative would be to use OOP (but perhaps a separate ticket should track making this sort of design change?):
```
class MyClassifier(Estimator):
    hyperparameters = [Hyperparameter(name='a_model_parameter', values=['a', 'b'], default='a')]
```

2. Let's find a way to support library-internal defaults too. In the case of catboost's learning rate, their internal default will be used when we don't provide a "learning_rate" parameter to the constructor. We should build functionality for doing so by default. One way to accomplish this would be to add a "LIBRARY_DEFAULT" setting:
```
class MyClassifier(Estimator):
    hyperparameter_ranges = {
        "a_model_parameter": {"values": ["option_a", "option_b], "default": Hyperparameter.LIBRARY_DEFAULT}
    }
```

When LIBRARY_DEFAULT is provided as the default value, the initial search point will not include the parameter in question, and our Pipeline and Component classes will not provide a value for that parameter to the underlying third-party transformer or estimator's constructor, to allow the third-party library to make its own determination for that parameter.

**Timeline**
To support something like this currently, I think we'd have to add code in autobase which generates the first set of parameters from the hyperparameter defaults, instead of calling `SKOptTuner.propose`. In #272 I'd like to move towards encapsulating that functionality in the Tuner classes, so perhaps this task should be part of that redesign.
